web/src/views/Pulls/Pull.tsx

```diff
@@ -76,27 +78,29 @@ const Pull = ({
       Icon: VisibilityRounded
     },
     {
-      id: 'changes',
-      text: 'Changes',
-      Icon: CodeRounded,
-      badgeValue:
-        diffs && diffs.value ? diffs.value.length : <CircularProgress size={12} color="inherit" />
+      id: clonesTabId,
+      text: 'Clones',
+      Icon: FileCopyRounded,
+      badgeValue: clones.value ? (
+          clones.value.length
+      ) : (
+          <CircularProgress size={12} color="inherit" />
+      )
     },
     {
-      id: 'compare',
+      id: compareTabId,
       text: 'Compare',
       Icon: CompareArrowsRounded
     },
     {
-      id: 'clones',
-      text: 'Clones',
-      Icon: FileCopyRounded,
-      badgeValue: clones.value ? (
-        clones.value.length
-      ) : (
-        <CircularProgress size={12} color="inherit" />
-      )
-    }
+      id: changesTabId,
+      text: 'Changes',
+      Icon: CodeRounded,
+      badgeValue: (
+          diffs && diffs.value ? diffs.value.length
+              : diffs && diffs.isFetching ? <CircularProgress size={12} color="inherit" /> : null
+      ),
```

Comment:
@vaddya pls give an advice how to achieve this in a batter manner

Reply:
No idea. If it works, that's fine.

Reply:
I mean, can i use normal if-else branching?

Reply:
You will have to return different components in these branches if you use if-else, so short answer is no.
Two `?:` is not that bad.

Reply:
Basic refactoring rule: `extract method` is best variant here imho